### PR TITLE
✨ Add branch.io sdk and referrer feature

### PR DESCRIPTION
**Pull request by williamchong007**
Added platform native sdk, working on react-native service/env

### CODE REVIEW

File: app/screens/referral-screen/referral-screen.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import { Button } from "../../components/button"
+import { Screen } from "../../components/screen"
+import { Sheet } from "../../components/sheet"
+import { Text } from "../../components/text"
+
+import { color, spacing } from "../../theme"
+
+import { logAnalyticsEvent } from "../../utils/analytics"
+
+export interface ReceiveScreenProps extends NavigationScreenProps {
```
Comment (nwingt) [inline app/screens/referral-screen/referral-screen.tsx:24]: `ReceiveScreenProps` -> `ReferralScreenProps`

File: app/screens/referral-screen/referral-screen.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+const BOTTOM_BAR: ViewStyle = {
+  alignItems: "center",
+  width: "100%",
+  padding: spacing[3],
+  backgroundColor: color.palette.white,
+}
+
+@inject((rootStore: RootStore) => ({
+  userStore: rootStore.userStore,
+  appReferralLink: rootStore.userStore.userAppReferralLink,
```
Comment (nwingt) [inline app/screens/referral-screen/referral-screen.tsx:71]: Seems injecting `appReferralLink` is redundant here as you can already retrieve it from userStore

**Comment by williamchong007** [comment]
@nwingt I have added a guard to prevent double handling of deeplink in the new commits, please check again
We can refactor to allow branchIo service to return how to handle branch link later, but for now I just block any handling
I have tested on android and ios simulator, and want to test on testflight, but probably need to wait for your analytics feature to finish testing first

File: app/models/root-store/root-store.ts [hunk 3]
```diff
@@ -61,15 +61,16 @@ export const RootStoreModel = types
...
-
-      if (URL_REGEX.test(url)) {
-        self.navigationStore.dispatch({
-          type: "Navigation/PUSH",
-          routeName: "ContentView",
-          params: {
-            content: self.readerStore.getContentByURL(url),
-          },
-        })
+      if (!self.env.branchIO.getIsClickedBranchLink) {
```
Comment (nwingt) [inline app/models/root-store/root-store.ts:64]: isnt `getIsClickedBranchLink` a function?

**Comment by nwingt** [comment]
@williamchong007 Please rebase

**Comment by williamchong007** [comment]
@nwingt rebased

**Comment by nwingt** [comment]
@williamchong007 Perhaps replace this with `likecoin:feature/branch-io`?

**Comment by williamchong007** [comment]
replaced by https://github.com/likecoin/likecoin-app/pull/162

**Pull request closed without merging**